packages/benchmark.rb

```diff
@@ -0,0 +1,52 @@
+require 'package'
+
+class Benchmark < Package
+  description 'A microbenchmark support library from Google'
+  homepage 'https://github.com/google/benchmark/'
+  version '1.5.2'
+  compatibility 'all'
+  source_url 'https://github.com/google/benchmark/archive/v1.5.2.tar.gz'
+  source_sha256 'dccbdab796baa1043f04982147e67bb6e118fe610da2c65f88912d73987e700c'
+
+  def self.patch
+    @limitsh = <<~EOF
+      --- a/src/benchmark_register.h
+      +++ b/src/benchmark_register.h
+      @@ -1,6 +1,7 @@
+       #ifndef BENCHMARK_REGISTER_H
+       #define BENCHMARK_REGISTER_H
+ 
+      +#include <limits>
+       #include <vector>
+ 
+       #include "check.h"
+    EOF
+    IO.write("limitsh.patch", @limitsh)
+    system 'patch -p 1 -i limitsh.patch'
+  end
+  
+  def self.prebuild
+    system "git clone git://github.com/google/googletest.git -b release-1.10.0 --depth 1" # Required for build, won't interfere with the gtest package
+  end
+    
+  def self.build
+    Dir.mkdir "builddir"
+    Dir.chdir "builddir" do
```

Comment:
Should this be?
```
Dir.chdir 'googletest' do
```
Where do you cd to the cloned repo?

Reply:
I don't cd into the cloned repo, it just goes into the source like a submodule and it's compiled for tests.

Reply:
So it clones to a folder the extracted tarball source can read already?

Reply:
Yes